code/modules/goonhub/api/routes/apiRoute.dm

```diff
@@ -27,7 +27,15 @@
 			if (islist(src.queryParams[key])) // Do we need to encode the value?
 				if (length(src.queryParams[key]) > 0)
 					for (var/subKey in src.queryParams[key])
-						.["[key]\[[subKey]\]"] = src.queryParams[key][subKey]
+						if (islist(src.queryParams[key][subKey]))
+							if (length(src.queryParams[key][subKey]))
+								for (var/idx in 1 to length(src.queryParams[key][subKey]))
+									// encode in array syntax e.x. ?&filters[servers][0]=main3&filters[servers][1]=main4
+									.["[key]\[[subKey]\]\[[idx-1]\]"] = src.queryParams[key][subKey][idx]
```

Comment:
Index is unnecessary here.

```suggestion
									.["[key]\[[subKey]\]\[\]"] = src.queryParams[key][subKey][idx]
```

Reply:
If an array has multiple elements you will lose all but the last one though

You will attempt to set multiple value to index "filters[servers][]"

Reply:
![image](https://github.com/goonstation/goonstation/assets/33204415/46afb4e0-251b-477f-b021-197330bbfe14)